tests/components/weatherkit/test_coordinator.py

```diff
@@ -4,30 +4,65 @@
 from unittest.mock import patch
 
 from apple_weatherkit.client import WeatherKitApiClientError
+from freezegun.api import FrozenDateTimeFactory
 
 from homeassistant.const import STATE_UNAVAILABLE
 from homeassistant.core import HomeAssistant
-from homeassistant.util.dt import utcnow
 
-from . import init_integration
+from . import init_integration, mock_weather_response
 
 from tests.common import async_fire_time_changed
 
 
-async def test_failed_updates(hass: HomeAssistant) -> None:
```

Comment:
I think it would be better if this was broken into multiple tests that target the specific scenarios. Perhaps something like:
1. test_failed_update_uses_stale_data_before_threshold
2. test_failed_update_becomes_unvailable_after_threhold
3. test_failed_update_recovers

Not not a blocker to moving this PR forward, IMO.

Reply:
Sounds like a good idea to me. Keeps the tests small and isolated